### ISSUE TITLE
[BUG] Upgrade: after draining the first node, the upgrade is stuck

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#1984]
**Describe the bug**

This was observed after https://github.com/harvester/harvester-installer/pull/229.

After pre-hook and drain of the first node are done, the upgrade is stuck because post-hook annotation is not set on plan secret.

The cluster has this condition:
```
  - lastUpdateTime: "2022-03-08T16:13:27Z"
    message: provisioning bootstrap node(s) custom-9441cf1c7106
    reason: Waiting
    status: Unknown
    type: Provisioned
```

And from rancher log:
```
2022/03/08 16:13:16 [INFO] Stopping metadata watch on apiextensions.k8s.io/v1, Kind=CustomResourceDefinition
2022/03/08 16:13:16 [INFO] Stopping metadata watch on rbac.authorization.k8s.io/v1, Kind=ClusterRole
2022/03/08 16:13:16 [INFO] Stopping metadata watch on rbac.authorization.k8s.io/v1, Kind=Role
2022/03/08 16:13:16 [INFO] Stopping metadata watch on rbac.authorization.k8s.io/v1, Kind=RoleBinding
2022/03/08 16:13:16 [INFO] Stopping metadata watch on admissionregistration.k8s.io/v1, Kind=MutatingWebhookConfiguration
2022/03/08 16:13:17 [INFO] [planner] rkecluster fleet-local/local: waiting: provisioning bootstrap node(s) custom-9441cf1c7106: waiting for probes: kube-controller-manager
2022/03/08 16:13:17 [INFO] [planner] rkecluster fleet-local/local: waiting: provisioning bootstrap node(s) custom-9441cf1c7106: waiting for probes: kube-controller-manager
2022/03/08 16:13:27 [INFO] [planner] rkecluster fleet-local/local: waiting: provisioning bootstrap node(s) custom-9441cf1c7106
2022/03/08 16:13:27 [INFO] [planner] rkecluster fleet-local/local: waiting: provisioning bootstrap node(s) custom-9441cf1c7106
2022/03/08 16:15:02 [INFO] Downloading repo index from http://harvester-cluster-repo.cattle-system/charts/index.yaml
2022/03/08 16:20:02 [INFO] Downloading repo index from http://harvester-cluster-repo.cattle-system/charts/index.yaml
2022/03/08 16:21:38 [INFO] [planner] rkecluster fleet-local/local: waiting: provisioning bootstrap node(s) custom-9441cf1c7106
2022/03/08 16:21:41 [INFO] [planner] rkecluster fleet-local/local: waiting: provisioning bootstrap node(s) custom-9441cf1c7106
2022/03/08 16:21:47 [INFO] [planner] rkecluster fleet-local/local: waiting: provisioning bootstrap node(s) custom-9441cf1c7106
```

**To Reproduce**
1. Create a 1.0.0 cluster
2. Upgrade with master ISO.

**Expected behavior**

**Support bundle**

**Environment:**
 - Harvester ISO version: 
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by bk201** [comment]
Rancher team is aware of this and is fixing it.

**Comment by bk201** [comment]
According to Donnie, this will be addressed in https://github.com/rancher/rancher/issues/36782 as well.

**Comment by bk201** [comment]
After upgrading Rancher from `v2.6.3-harvester1` to `v2.6-5bb08b3692ac3c18ec1361694a095e2c25eb0346-head`, I can confirm. I can trigger the RKE2 upgrade cycle with drain hooks enabled.

**Comment by bk201** [comment]
The Rancher version is bumped and the issue is supposed to be fixed.
It can be tested once the upgrade feature is ready.

**Comment by TachunLin** [comment]
Verified **fixed** on one click upgrade to v1.0.1-rc2. Close this issue.

### Result
1. Can upgrade the 3 nodes harvester to v1.0.1-rc2 successfully without stuck
  ![image](https://user-images.githubusercontent.com/29251855/162128942-7099b45a-6e4b-4a2b-b5c3-aa188d8f6040.png)

2. Can proceed to upgrade the second node after draining the first node  
  ![image](https://user-images.githubusercontent.com/29251855/162128810-bc8f5c41-cc38-4118-a280-0951cd7eb6d4.png)
 

### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: v1.0.0 GA to v1.0.1-rc2

### Verify Steps
1. Prepare v1.0.0 three nodes harvester cluster on Provo machines
3. Create a ubuntu and SLES JeOS image
4. Create a ubuntu and SLES virtual machine
5. Enable network to `harvester-mgmt`
6. Create a virtual network `vlan1` with id `1`
7. Setup back target to AWS S3
8. Create two backups for the ubuntu and SLES VM
9. Shutdown all VMs
10. Follow the steps to upgrade to v1.0.1-rc2
https://github.com/harvester/docs/blob/42598035cfe79856f7e159d4c47c4d7d29178900/docs/upgrade/automatic.md
11. Check the one click upgrade completely finished
12. SSH to management node 
13. Run `sudo -i kubectl get managedcharts harvester -n fleet-local -o yaml | yq e '.spec.values' - `
14. Check `kube-vip.config` not exists
15. Check `longhorn.defaultDataPath: /var/lib/harvester/defaultdisk` exists 

### Additional Context
* Support Bundle
[supportbundle_138129b2-b170-4944-bcae-bc40ff88707e_2022-04-07T05-48-23Z.zip](https://github.com/harvester/harvester/files/8439975/supportbundle_138129b2-b170-4944-bcae-bc40ff88707e_2022-04-07T05-48-23Z.zip)
